### PR TITLE
add rank column in leaderboard

### LINKED ISSUE #251: Show Rank and "Your Rank" on leaderboard
- [x] Show Ranks on leaderboard.
- [x] If user is logged in show "Your Rank" on top of leaderboard and then the reset of the leaderboard. Similar to the leaderboard shown on [Hackerblocks](https://hack.codingblocks.com/app/)

### PR DESCRIPTION
**Pull request by prabalsingh24**
fixes #251 
Screenshot_2020-05-19 BOSS(1)

### CODE REVIEW

**Comment by prabalsingh24** [comment]
@hereisnaman Have a look at this whenever you're free :)

**Comment by hereisnaman** [comment]
Rather than showing the rank separately, if the user stats are not already in the table, add a row at the top in the table with the user's rank and other stats. Highlight the row for the logged in user stat using different bg.

**Comment by prabalsingh24** [comment]
When user column in the table : 
Screenshot_2020-05-19 BOSS(3)

When user column is not in the table : 

Screenshot_2020-05-19 BOSS(2)

When user is not logged in or is NOT int the leaderboard at all 

Screenshot_2020-05-19 BOSS(4)

**Comment by prabalsingh24** [comment]
@hereisnaman Have a look :)

**Comment by hereisnaman** [comment]
> @hereisnaman Have a look :)

Use `#fcfff5` as bg color

**Comment by prabalsingh24** [comment]
Screenshot_2020-05-19 BOSS(5)

This color is very light. You sure?

**Comment by prabalsingh24** [comment]
@hereisnaman 
Is this okay? 
Screenshot_2020-05-19 BOSS(5)

**Comment by prabalsingh24** [comment]
@hereisnaman the `#fcfff5` colour is almost colourless. You sure this should be the colour?

**Comment by prabalsingh24** [comment]
@hereisnaman have a look at this too :)

File: routes/root.js [hunk 1]
```diff
@@ -57,13 +57,30 @@ route.get('/leaderboard/:year?', (req, res) => {
...
       const lastPage = Math.ceil(count / options.size)
-
+      const showUserAtTop = loggedInUser.user && !rows.some(row => row.user === loggedInUser.user)
+      rows.forEach((row) => {
+          if(githubDetails && githubDetails.username === row.user){
+              row.color = "#fcfff5"
+          }
+          else {
+              row.color = null;
+          }
```
Comment (hereisnaman) [inline routes/root.js:82]: Move this logic to the view.

**Comment by hereisnaman** [comment]
> @hereisnaman have a look at this too :)

Move the colors logic to view. Otherwise else is fine.

**Comment by prabalsingh24** [comment]
@hereisnaman Because ``` {{#if loggedInUser.user === user.user}} {{/if}}``` can't be done in handlebars, I can't think of an easy solution.
I can think of following ways to do this:
1. By creating a helper block. Like this - [stackoverflow-answer](https://stackoverflow.com/questions/8853396/logical-operator-in-a-handlebars-js-if-conditional)
2. Have a `user.isColoured` boolean value. ( but again the logic  will be same and will be on the backend only)
3. Leave it as it is.
Let me know what you think :)

**Comment by hereisnaman** [comment]
> @hereisnaman Because ` {{#if loggedInUser.user === user.user}} {{/if}}` can't be done in handlebars, I can't think of an easy solution.
> I can think of following ways to do this:
> 
>     1. By creating a helper block. Like this - [stackoverflow-answer](https://stackoverflow.com/questions/8853396/logical-operator-in-a-handlebars-js-if-conditional)
> 
>     2. Have a `user.isColoured` boolean value. ( but again the logic  will be same and will be on the backend only)
> 
>     3. Leave it as it is.
>        Let me know what you think :)

Pass `isColored` and attach a class based on that. In css add bg color for that class.

**Comment by prabalsingh24** [comment]
@hereisnaman Done :)

**Pull request merged**